providers/volume_group.rb

```diff
@@ -0,0 +1,216 @@
+      'hotspare_y' => [(@new_resource.hotspare.eql? "y") && (!vginfo[15].eql? "1"), "-h #{@new_resource.hotspare}"],
+      'hotspare_Y' => [(@new_resource.hotspare.eql? "Y") && (!vginfo[15].eql? "2"), "-h #{@new_resource.hotspare}"],
+      'activate_on_boot' => [(@new_resource.activate_on_boot.eql? 'yes') && (!vginfo[10].eql? '1'), '-a y'],
+      'activate_on_boot_off' => [(@new_resource.activate_on_boot.eql? 'no') && (!vginfo[10].eql? '0'), '-a n'],
+      'lost_quorom_varyoff_off' => [(@new_resource.lost_quorom_varyoff.eql? "no") && (!vginfo[9].eql? '0'), '-Q n'],
+      'lost_quorom_varyoff_on' => [(@new_resource.lost_quorom_varyoff.eql? "yes") && (!vginfo[9].eql? '1'), '-Q y'],
+      'factor' => [(!@new_resource.factor.nil?) && (!vginfo[21].eql? "2") && (!@new_resource.factor.eql?  vginfo[7].to_i / 1016), "-t #{@new_resource.factor}"],
+      'big' => [(@new_resource.big) && (vginfo[21] != "1"), '-B'],
+      'force' => [(@new_resource.force), '-f'],
+    }
+    # Assign options
+    option_map.each do |_opt, val|
+      if val[0]
+        chvg = chvg << " #{val[1]}"
+        usable_options = true
+      end
+    end
+
+
+    # Any other options passed-in
+    unless @new_resource.options.nil?
+      usable_options = true
+      @new_resource.options.each do |o, v|
+          chvg = chvg << " #{o}"
+          if !v.nil? || !v.empty?
+             chvg = chvg << " #{v}"
+          end
+      end
+    end
+    
+
+    chvg = chvg << " #{@new_resource.name}"
+
+    # Perform chvg
+    if usable_options
+       converge_by("chvg: changing #{@new_resource.name} : #{chvg}") do
+         Chef::Log.info("chvg: chaning #{@new_resource.name} : ${chvg}")
+         chvg_do = Mixlib::ShellOut.new(chvg)
+         chvg_do.valid_exit_codes = 0
+         chvg_do.run_command
+         chvg_do.error!
+         chvg_do.error?
+       end
+    end
+  end
+end
+
+def best_fit_disk(size)
+  # Nil by default
+  disk = nil
+  # Get unused disk array
+  unused_disks = unused_disk_search
+  Chef::Log.info("unused disks: #{unused_disks}")
+  # Interate through array to find size of disk, keep if larger than size requested
+  unless unused_disks.nil?
+    disk_choices = {}
+    unused_disks.each do |this_disk|
+      disk_name = this_disk
+      Chef::Log.info("mkvg: examining disk size for #{this_disk}")
+      bootinfo = Mixlib::ShellOut.new("bootinfo -s #{this_disk}")
+      bootinfo.run_command
+      disk_size = bootinfo.stdout.to_i / 1024
+      if disk_size >= size
+        disk_choices[disk_name] = disk_size
+      end
+    end
+    Chef::Log.info("mkvg: disk sizes: #{disk_choices}")
+    disk_chosen = disk_choices.min_by { |_k, v| v }
+    Chef::Log.info("mkvg: disk chosen: #{disk_chosen}")
+    unless disk_chosen.nil?
+      disk = disk_chosen[0]
+    end
+  end
+  disk
+end
+
+def unused_disk_search
+  # Find unused disk and put into array
+  unused_disk = Mixlib::ShellOut.new("lspv | awk '{if ($3 == \"None\") print $1;}' |  tr '\n' ' '")
```

Comment:
Avoid multiple pipes in ShellOut,. Maybe you can do the same thing with a single pipe lspv | awk '$3 == "None" {printf "%s ",$1}'